etl/steps/grapher/worldbank_wdi/2022-05-26/wdi.py

```diff
@@ -0,0 +1,304 @@
+import re
+import json
+import zipfile
+from typing import Iterable, Optional, List, Union
+from pathlib import Path
+
+import structlog
+import pandas as pd
+
+from owid.catalog import Dataset, Table, Source, VariableMeta
+from owid.catalog.utils import underscore
+from owid.walden import Catalog
+from etl.paths import DATA_DIR
+from etl import grapher_helpers as gh
+from etl.db import get_connection
+
+log = structlog.get_logger()
+
+
+def get_grapher_dataset() -> Dataset:
+    version = Path(__file__).parent.stem
+    fname = Path(__file__).stem
+    namespace = Path(__file__).parent.parent.stem
+    dataset = Dataset(DATA_DIR / f"garden/{namespace}/{version}/{fname}")
+    # short_name should include dataset name and version
+    dataset.metadata.short_name = (
+        f"{dataset.metadata.short_name}__{version.replace('-', '_')}"
+    )
+
+    return dataset
+
+
+def get_grapher_tables(dataset: Dataset) -> Iterable[Table]:
+    fname = Path(__file__).stem
+    table = dataset[fname]
+    assert len(table.metadata.primary_key) == 2
+    var_codes = table.columns.tolist()  # type: ignore
+
+    # retrieves raw data from walden
+    version = Path(__file__).parent.stem
+    fname = Path(__file__).stem
+    namespace = Path(__file__).parent.parent.stem
+    walden_ds = Catalog().find_one(
+        namespace=namespace, short_name=fname, version=version
+    )
+    local_file = walden_ds.ensure_downloaded()
+    zf = zipfile.ZipFile(local_file)
+    df_vars = pd.read_csv(zf.open("WDISeries.csv"))
+    df_vars.dropna(how="all", axis=1, inplace=True)
+    df_vars.columns = df_vars.columns.map(underscore)
+    df_vars.rename(columns={"series_code": "indicator_code"}, inplace=True)
+    df_vars["indicator_code"] = df_vars["indicator_code"].apply(underscore)
+    df_vars = df_vars.query("indicator_code in @var_codes").set_index(
+        "indicator_code", verify_integrity=True
+    )
+
+    df_vars["indicator_name"].str.replace(r"\s+", " ", regex=True)
+    clean_source_mapping = load_clean_source_mapping()
+
+    # construct metadata for each variable
+    vm = VariableMatcher()
```

Comment:
Just a note here - sources other than files (like external database) are not gonna change checksum of this step. So if someone changes metadata in grapher DB, this dataset won't be refreshed.

(I don't think it matters here because we don't have grapher checksums yet and it's unlikely that this would happen for WDI)

Reply:
ah good point. Ya hopefully this is unlikely to ever actually cause a problem. Once this version is in the `live` grapher, no one should be updating metadata on old versions of the datasets.